web/lib/api/cache/stats.ts

```diff
@@ -66,12 +66,13 @@ export async function getModelMetricsClickhouse(
   });
   const query = `
   select model, 
+    provider,
     sum(completion_tokens) as sum_completion_tokens, 
     sum(prompt_tokens) as sum_prompt_tokens, 
     sum(completion_tokens + prompt_tokens) as sum_tokens 
   from cache_hits 
   where (${builtFilter.filter})
-  group by model`;
+  group by model, provider`;
```

Comment:
why are we grouping by provider here also?

Reply:
for cache total savings we need provider to calculate the cost accurately now

```
async function handler({
  req,
  res,
  userData: { orgId },
}: HandlerWrapperOptions<Result<number, string>>) {
  res
    .status(200)
    .json(
      resultMap(await getModelMetricsClickhouse(orgId, "all"), (modelMetrics) =>
        modelMetrics.reduce(
          (acc, modelMetric) => acc + modelCost(modelMetric),
          0
        )
      )
    );
}

```